miprometheus/problems/question_context_to_class/babiqa_dataset_single_question.py

```diff
@@ -0,0 +1,514 @@
+            for answer in story:
+                a = answer.lower()
+                if a not in answ_to_ix:
+                    ix = len(answ_to_ix)
+                    answ_to_ix[a] = ix
+                    itos_d.append(a)
+                    # print(a,ix)
+
+            #go through all the answers
+            for answer in answers:
+                a = answer.lower()
+                if a not in answ_to_ix:
+                    ix = len(answ_to_ix)
+                    answ_to_ix[a] = ix
+                    itos_d.append(a)
+                    # print(a, ix)
+
+        #return the corresponding dictionnaries
+        ret = (answ_to_ix)
+        return ret, itos_d
+
+    def build_dictionaries(self):
+
+        """Creates the word embeddings BABI QA 
+
+        - 1. Collects all datasets word
+        - 2. Uses Language object to create the embeddings
+
+         If it is the first time you run this code, it will take longer to load the embedding from torchtext
+         """
+
+        print(' ---> Constructing the dictionaries with word embedding, may take some time ')
+
+        # making an empty list of words meant to store all possible datasets words
+        text = []
+        tasks = self.tasks
+
+        data = self.load_data(tasks=tasks, tenK=self.tenK, add_punctuation=True, data_type='train', outmod="embedding")
+        data = data + self.load_data(tasks=tasks, tenK=self.tenK, add_punctuation=True, data_type='valid',
+                                     outmod="embedding")
+        data = data + self.load_data(tasks=tasks, tenK=self.tenK, add_punctuation=True, data_type='test',
+                                     outmod="embedding")
+
+        # make placeholders dictionnaries with special caracters
+        answ_to_ix = {".": 0, "?": 1, "_": 2}
+        itos_d = [".", "?", "_"]
+
+        # load all words from training data to a list named words_list[]
+        self.fix_length = 0
+        for q in tqdm(data):
+            # display a progress bar
+            story, answers = q
+            self.fix_length = max(self.fix_length, len(story))
+            for word in story:
+                text.extend([word.lower()])
+
+            # go through all the stories
+            for answer in story:
+                a = answer.lower()
+                if a not in answ_to_ix:
+                    ix = len(answ_to_ix)
+                    answ_to_ix[a] = ix
+                    itos_d.append(a)
+                    # print(a,ix)
+
+            # go through all the answers
+            for answer in answers:
+                a = answer.lower()
+                if a not in answ_to_ix:
+                    ix = len(answ_to_ix)
+                    answ_to_ix[a] = ix
+                    itos_d.append(a)
+                    # print(a, ix)
+
+        """ build embeddings from the chosen database / Example: glove.6B.100d """
+
+        self.language.build_pretrained_vocab(text, vectors=self.embedding_type, tokenize=self.tokenize)
```

Comment:
regarding the problem with the size of embeddings...

I think that language (btw. I think this is a bad name, means exactly nothing, if I understand it correctly it is... TextEncoder?) should set and return its value (i.e. size of embeddings)...